Add missing rpc.shutdown()

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44418 [WIP] Use streams from pool on RPC callees
* #44396 Use TP Tensor.metadata to carry device info
* **#44417 Add missing rpc.shutdown()**

Differential Revision: [D23626208](https://our.internmc.facebook.com/intern/diff/D23626208)

mrshenli committed:
Add missing rpc.shutdown()

[ghstack-poisoned]

pritamdamania87 reviewed (approved):
Was this test flaky/failing due to the missing shutdown before?

mrshenli committed:
Update on "Add missing rpc.shutdown()"


Differential Revision: [D23626208](https://our.internmc.facebook.com/intern/diff/D23626208)

[ghstack-poisoned]
